src/triton_gazebo/src/thruster_driver_plugin.cpp

```diff
@@ -80,7 +71,9 @@ namespace triton_gazebo
         thruster.push_back(_model->GetLink("triton_auv::thruster5::thruster"));
         thruster.push_back(_model->GetLink("triton_auv::thruster6::thruster"));
```

Comment:
Populating the LinkPtr vector needs to be done in a loop using the thruster_count variable. right now were taking in an arbitrary array size but can only drive 6 anyways. 
replace all these push_backs with 
`for (int i = 0; i < this->thruster_count; i++)`
`{`
`    std::string thruster_name = (std::string)"triton_auv::thruster" + to_string(i) + (std::string)"::thruster";`
`    thruster.push_back(_model->GetLink(thruster_name));`
`}`